src/app/components/keyfilter/keyfilter.ts

```diff
@@ -227,8 +231,9 @@ export class KeyFilter implements Validator {
         }
 
         let valueCheck = this.el.nativeElement.value || '';
-
-        let val = valueCheck + cc;
+        const selectionStart = (<HTMLInputElement>e.currentTarget).selectionStart || 0;
+        const selectionEnd = (<HTMLInputElement>e.currentTarget).selectionEnd || 0;
+        let val = valueCheck.substring(0, selectionStart) + cc + valueCheck.substring(selectionEnd);
```

Comment:
I have not verified this but should lines 234 and 235 
` || 0`
be  replaced with
`|| Number.MAX_SAFE_INTEGER`